### ISSUE TITLE
corpus: second scan found 5 suspicious clusters, incl. 2 errors on nanoGPT/train.py

### ISSUE THREAD

**Issue by HimanshuJanbandhu** [highwaterlabs/torch-preflight#58]
Second corpus, chosen to stress different shapes than the first seven: `nanoGPT`, `minGPT`, `mosaicml/composer`, `huggingface/peft`, `OpenRLHF`, `axolotl`.

**1,829 files, 406 findings — 0.222/file**, closely matching the first corpus (0.20/file), so the rules behave consistently on unseen code.

| repo | files | err | warn | note |
|---|---|---|---|---|
| nanoGPT | 15 | 2 | 1 | 0 |
| minGPT | 9 | 1 | 0 | 0 |
| composer | 392 | 8 | 8 | 235 |
| peft | 437 | 43 | 13 | 59 |
| OpenRLHF | 65 | 0 | 1 | 2 |
| axolotl | 911 | 8 | 0 | 25 |

### 1. `nanoGPT/train.py` — 2 TG001 errors, and neither is training code

```
167 │ model_args[k] = checkpoint_model_args[k]            # config ints from a checkpoint
177 │ state_dict[k[len(unwanted_prefix):]] = state_dict.pop(k)   # key renaming
```

Both are **checkpoint surgery**. The first moves config integers between dicts; the second renames `state_dict` keys on freshly-loaded leaf tensors that have no graph. We report `CRITICAL_OOM` on the most widely-read minimal training script there is, which makes this the highest-priority item here.

### 2. `peft/tests/` — 29 TG002 errors

The docs claim pytest `test_*` functions are exempt, but these fire. Samples:

- `tests/test_boft.py:36`, inside `def test_boft_state_dict(self, tmp_path)` — a test **method on a class**, not a module-level function.
- `tests/testing_common.py:551`, inside `def _test_merge_layers_nan(...)` — a helper named `_test_*`, called by real tests.

Hypothesis: the exemption only matches top-level `test_*` functions. Unverified.

### 3. `composer/composer/loss/loss.py` — 2 findings in library code

- `:109` TG005 on `F.cross_entropy(input=input, ...)`. The only `log_softmax` in that function is in the **mutually exclusive `else` branch** (`xentropy = -(target * F.log_softmax(input, dim=1))`), where `input` is the *argument to* softmax, not its output.
- `:60` TG006 on `F.binary_cross_entropy_with_logits(input=input, ...)`, which is correct usage.

### 4. `composer/tests/common/models.py` — 6 TG005 warnings

`nn.Sequential(..., torch.nn.Softmax(dim=-1))` with cross-entropy. These look **true** — the shape TG005 exists for — but they are dummy models in test fixtures.

### 5. `peft/src/peft/tuners/lora/conversion.py:326,327,333,334` — 4 TG001 errors in library code

Not yet read.

## Important: reduction did not reproduce

I reduced clusters 1 and 2 to minimal fixtures and **all came back clean**, so my hypotheses about the causes are wrong or incomplete — the behaviour depends on wider file context. nanoGPT's `train.py` is module-level script code where everything shares one scope, which is a plausible direction but unverified.

**Do not fix from the descriptions above.** Reproduce against the real file first:

```bash
git clone --depth 1 https://github.com/karpathy/nanoGPT
torch-preflight check nanoGPT/train.py
```

A fix without a red test is not verifiable here — see #46, where a plausible-looking change silenced a real leak and neither the test suite nor a 1,615-file scan could see it.
